Lib/asyncio/queues.py

```diff
@@ -163,21 +207,34 @@ async def get(self):
                     self._getters.remove(getter)
                 except ValueError:
                     # The getter could be removed from self._getters by a
-                    # previous put_nowait call.
+                    # previous put_nowait call,
+                    # or a shutdown call.
                     pass
                 if not self.empty() and not getter.cancelled():
                     # We were woken up by put_nowait(), but can't take
                     # the call.  Wake up the next in line.
                     self._wakeup_next(self._getters)
                 raise
+            if self._is_shutdown_immediate():
+                raise QueueShutDown
         return self.get_nowait()
 
     def get_nowait(self):
         """Remove and return an item from the queue.
 
         Return an item if one is immediately available, else raise QueueEmpty.
+
+        Raise a QueueShutDown if _shutdown_state is set to
+        `SHUTDOWN_IMMEDIATE`.
+
+        Raise a QueueShutDown if _shutdown_state is set to
+        `SHUTDOWN` and queue is empty.
```

Comment:
```suggestion
        Raises QueueShutDown if the queue has been shut down and is empty, or
        if the queue has been shut down immediately.
```

Doc-strings are for humans